src/Streams.cpp

```diff
@@ -55,18 +59,32 @@ void Streams::checkStreamStatus() {
                        StreamPointers.end());
 }
 
-void Streams::add(std::shared_ptr<Stream> s) { StreamPointers.push_back(s); }
+json Streams::getStreamStatuses() {
```

Comment:
It seems to me that the output of this function is never actually tested. If was more petty I would ask you to write a unit test for this. I am willing to let it go though.